digitalocean/datasource_digitalocean_regions.go

```diff
@@ -0,0 +1,188 @@
+package digitalocean
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var (
+	dataSourceDigitalOceanRegionsFilterKeys = []string{
+		"slug",
+		"name",
+		"available",
+		"features",
+		"sizes",
+	}
+
+	dataSourceDigitalOceanRegionsSortKeys = []string{
+		"slug",
+		"name",
+		"available",
+	}
+)
+
+func dataSourceDigitalOceanRegions() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDigitalOceanRegionsRead,
+		Schema: map[string]*schema.Schema{
+			"filter": filterSchema(dataSourceDigitalOceanRegionsFilterKeys),
+			"sort":   sortSchema(dataSourceDigitalOceanRegionsSortKeys),
+
+			"regions": {
+				Type:     schema.TypeSet,
```

Comment:
```suggestion
				Type:     schema.TypeList,
```

I think we want to make this a list rather than a set. Since we can sort the list, the order does matter. It also has the benefit of allowing you to reference a specific value more easily as it is indexed predictably. Eg:

```
output "slug" {
  value = data.digitalocean_regions.with-storage.regions.0.slug
}
```

https://www.terraform.io/docs/extend/schemas/schema-types.html#typelist

Reply:
Makes sense. Will implement.

Reply:
Pushed the change plus a test helper.